perun-wui-core/src/main/java/cz/metacentrum/perun/wui/model/beans/ApplicationFormItem.java

```diff
@@ -238,16 +270,25 @@ public final native void setOrdnum(int ordnum) /*-{
 	 *
 	 * @param applicationTypes List of application types this form item is used on.
 	 */
-	public final void setApplicationTypes(ArrayList<Application.ApplicationType> applicationTypes) {
-
-		// TODO - implementation
-
+	public final void setApplicationTypes(List<Application.ApplicationType> applicationTypes){
+		JSONArray array = new JSONArray();
+		for(int i = 0; i < applicationTypes.size(); i++){
+			if(applicationTypes.get(i).equals(Application.ApplicationType.EXTENSION)){
```

Comment:
You can use here `array.set(i, new JSONString(applicationTypes.get(i).toString.toUpper()))` instead the whole if - else branches. This has also the advantage that if some new values are added, this code will support them as well.